pkg/asset/installconfig/openstack/validation/cloudinfo.go

```diff
@@ -226,20 +228,24 @@ func (ci *CloudInfo) collectInfo(ic *types.InstallConfig, opts *clientconfig.Cli
 
 	return nil
 }
-
-func (ci *CloudInfo) getSubnet(subnetID string) (*subnets.Subnet, error) {
-	if subnetID == "" {
-		return nil, nil
-	}
-	subnet, err := subnets.Get(ci.clients.networkClient, subnetID).Extract()
-	if err != nil {
-		if isNotFoundError(err) {
-			return nil, nil
+func (ci *CloudInfo) getSubnets(controlPlanePort *openstack.PortTarget) ([]*subnets.Subnet, error) {
+	controlPlaneSubnets := make([]*subnets.Subnet, 0, len(controlPlanePort.FixedIPs))
+	for _, fixedIP := range controlPlanePort.FixedIPs {
+		page, err := subnets.List(ci.clients.networkClient, subnets.ListOpts{ID: fixedIP.Subnet.ID, Name: fixedIP.Subnet.Name}).AllPages()
```

Comment:
Yes. In case the user is still using MachinesSubnet, we want to allow it to be find and the user does not provide any info about the Network with MachinesSubent.